File: src/gestorAplicacion/Restaurantes/Mesa.java
package gestorAplicacion.Restaurantes;

import java.io.Serializable;

import gestorAplicacion.Personas.GrupoHuespedes;

public class Mesa implements Serializable {
    private boolean ocupado;
    private GrupoHuespedes dueños; // quienes tienen reservada esta mesa
    private int id;

    public Mesa(int id) {
        this.id = id;
    }

    public void asignarDueños(GrupoHuespedes grupo) {
        this.dueños = grupo;
        dueños.setMesaReservada(this);
        ocupado = true;
    }

    public void vaciarMesa() {
        dueños.setMesaReservada(null);
        this.dueños = null;
        this.ocupado = false;
    }


    @Override
    public String toString() {
        if (ocupado == true) {
            return "Mesa: " + id + ", Reservada por " + dueños.toString();
        }
        return "Disponible";
    }

    // * Getters y setters */
    public boolean isOcupado() {
        return ocupado;
    }

    public void setOcupado(boolean ocupado) {
        this.ocupado = ocupado;
    }

    public GrupoHuespedes getDueños() {
        return dueños;
    }

    public void setDueños(GrupoHuespedes dueños) {
        this.dueños = dueños;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

}

File: python/src/gestor_aplicacion/restaurantes/Mesa.py
from gestor_aplicacion.personas.GrupoHuespedes import GrupoHuespedes


class Mesa:
    def __init__(self, id):
        self.ocupado = False
        self.dueños:GrupoHuespedes = None  # quienes tienen reservada esta mesa
        self.id = id

    def asignarDueños(self, dueños:GrupoHuespedes ):
        self.dueños = dueños
        dueños.set_mesa_reservada(self)
        self.ocupado = True

    def vaciarMesa(self,dueños:GrupoHuespedes):
        dueños.set_mesa_reservada(None)
        self.dueños = None
        self.ocupado = False

    def __str__(self):
        if self.ocupado:
            return "Mesa: " + str(self.id) + ", Reservada por " + str(self.dueños)
        return "Disponible"

    # Getters y setters
    def isOcupado(self):
        return self.ocupado

    def setOcupado(self, ocupado):
        self.ocupado = ocupado

    def getDueños(self):
        return self.dueños

    def setDueños(self, dueños):
        self.dueños = dueños

    def getId(self):
        return self.id

    def setId(self, id):
        self.id = id
